nomenclature/__init__.py

```diff
@@ -109,8 +107,12 @@ def validate(df):
     msg = 'The following {} are not defined in the nomenclature:\n    {}'
     for col, codelist, ext in cols:
         invalid = [c for c in df.data[col].unique() if c not in codelist]
+        for i in reversed(invalid):
```

Comment:
please add a comment here to explain what this section is doing.

also, why is this `reversed`? the following should also just work
```suggestion
        for i in invalid:
```

Reply:
I had minor difficulties with the remove command and iteration through the list. From my point of view, it is a _classic_ way to iterate through the list from behind and remove all valid entries from the invalid list. However, I will add more comments for clarification.